api/src/main/java/io/minio/ComposeSource.java

```diff
@@ -153,56 +152,52 @@ public void buildHeaders(long objectSize, String etag) throws IllegalArgumentExc
                 + objectSize);
       }
     }
+  }
 
-    Map<String, String> headers = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
-    headers.put("x-amz-copy-source", S3Escaper.encodePath(bucketName + "/" + objectName));
-    headers.put("x-amz-copy-source-if-match", etag);
-
-    if (headerMap != null) {
-      headers.putAll(headerMap);
+  /** Argument builder of {@link ComposeSource}. */
+  public static final class Builder extends ObjectVersionArgs.Builder<Builder, ComposeSource> {
+    @Override
+    protected void validate(ComposeSource args) {
```

Comment:
Since this method is not doing anything apart from calling the super method, it is not required anymore.